addon/serializers/application.js

```diff
@@ -190,8 +190,8 @@ export default DS.RESTSerializer.extend({
       hasMany.forEach( function( child ) {
         json[key].objects.push({
           '__type'    : 'Pointer',
-          'className' : child.parseClassName(),
-          'objectId'  : child.get( 'id' )
+          'className' : _this.parseClassName(child.type.typeKey),
```

Comment:
This is something hopefully @joshfester or @Kira2 can speak to.  I'm honestly just going through the commits at github.com/joshfester/ember-parse-adapter and extracting the ones that are applicable to Ember Data 1.0.0-beta.15 and replaying them onto the Ember CLI Add-on version of the adapter.  I will then do the same thing with other commits against Ember Data 1.0.0-beta.16 and again with other generic fixes, etc.

I have not been focusing on the specific changes that were applied or why - if the tests are passing and the demo app still functions I have viewed it as being functionally correct.  I am trying to facilitate the convergence of several different development efforts into this single repository in the Ember CLI format.  At that point I, and others, can begin focusing specifically on refactors and improvements.

If there is a different approach you would rather I and others to follow I am open to suggestions - I am just trying to move progress forward.